File: 릿코드/1122-relative-sort-array/1122-relative-sort-array.js
/**
 * @param {number[]} arr1
 * @param {number[]} arr2
 * @return {number[]}
 */
var relativeSortArray = function(arr1, arr2) {
    const arr = arr1.filter((v) => !arr2.includes(v)).sort((a, b) => a-b);
    return [...arr2.map(v => arr1.filter(v2 => v==v2)).flat(), ...arr];
};

File: 릿코드/1122-relative-sort-array/1122-relative-sort-array.java
class Solution {
    public int[] relativeSortArray(int[] arr1, int[] arr2) {
        int[] index = {0};
        Set<Integer> arr2Set = new HashSet<>(){{
           for(int arr : arr2) add(arr);
        }};
        List<Integer> remainder = new ArrayList<>(){{
            for(int arr : arr1) if(!arr2Set.contains(arr)) add(arr);
        }};

        int[] result = new int[arr1.length];

        for(int arr : arr2) {
            for(int ar : arr1) {
                if(ar == arr) result[index[0]++] = arr;
            }
        }

        remainder.stream().sorted().forEach(remain -> result[index[0]++] = remain);

        return result;
    }
}